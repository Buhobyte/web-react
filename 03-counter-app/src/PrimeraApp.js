import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

// const PrimeraApp = (props) => {
// const PrimeraApp = ({ saludo="Valor por defecto" }) => {
const PrimeraApp = ({ saludo,subtitulo }) => {

    return (
        <>
            <h1> {saludo}</h1 >
            <h1> {subtitulo}</h1 >
            <p> Parrago de mi div</p>
        </>

    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}

PrimeraApp.defaultProps ={
    subtitulo: "Soy un sibtitulo"
}

export default PrimeraApp;