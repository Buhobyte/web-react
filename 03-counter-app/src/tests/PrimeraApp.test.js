import React from 'react';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';

describe('Prueba de PrimeraAPP', () => {
    // test('Debe mostrar el mensaje Hola Soy Gokuw', () => {
    //     const saludo = "Hola Soy Goku";
    //     const { getByText } = render(<PrimeraApp saludo={saludo} />);

    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('debe mostrar <PrimeraApp/> correctamente', () => {
        const saludo = 'Hola, Soy Goku';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el subitulo enviado por props', () => {
        const saludo = 'Hola, Soy Goku';
        const subtitulo = 'Soy un subtitulo';

        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

        const textoParrafo = wrapper.find('#idSubtitulo').text();
        console.log(textoParrafo);

        expect(textoParrafo).toBe(subtitulo);

    })


})
