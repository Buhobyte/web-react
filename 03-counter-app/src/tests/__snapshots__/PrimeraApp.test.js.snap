// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de PrimeraAPP debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, Soy Goku
     !!!
  </h1>
  <h1
    id="idSubtitulo"
  >
    Soy un subtitulo
  </h1>
  <p>
     Parrago de mi div
  </p>
</Fragment>
`;
